network/iptables.go

```diff
@@ -100,21 +99,74 @@ func ipTablesRulesExist(ipt IPTables, rules []IPTablesRule) (bool, error) {
 	return true, nil
 }
 
+func ipTablesCleanAndBuild(ipt IPTables, rules []IPTablesRule) (IPTablesRestoreRules, error) {
+	tablesRules := IPTablesRestoreRules{}
+
+	// Build delete rules
```

Comment:
Is there any way to ensure the order is wrong before just automatically deleting/creating the rules?

Reply:
I think it's possible if we have before memorized the order, but I'm not sure we can in every case memorize the order. And we will need to store this state outside the memory (ie: etcd) for the crash case. 
So I'm not sure it's a real benefit compared to the complexity it's will add.